doc/syntax.rst

```diff
@@ -395,3 +437,50 @@ We can now call the ``StateTeleportation`` subroutine, with ``sq=1``,
 and apply it to modes 0, 2, and 3.
 
 .. note:: Make sure to avoid **circular includes** when using the ``include`` statement.
+
+
+Program types
+-------------
+
+A program type can be define with the ``type`` keyword in the metadata. The type
+includes support for a specific set of experiments and might differ in the way
+that they are defined inside a Blackbird script. Currently, the only supported
+type is ``tdm``, which stands for time-domain multiplexing, and runs a photonic
+quantum circuit in the time domain encoding.
+
+Time-domain multiplexing (TDM)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To define a TDM program you can declare the ``tdm`` type in the metadata
+together with two different keyword arguments:
+
+.. code-block:: python
+
+    type tdm (temporal_modes=42, copies=1000)
+
+The TDM program requires a set of gates that is looped over a number of times
+equal to the number of temporal modes, which is defined in the type options. The
+set of gates only needs to be defined one time, accompanied by arrays containing
+the parameters that are to be used in each loop.
+
+TDM programs has reserved words starting with a ``p`` followed by a number; e.g.,
+``p0``, ``p1``, or ``p42``. These are placeholders for the parameters in their
+corresponding arrays (see script example below).
```

Comment:
What happens if you _don't_ use `p`? E.g.,

```python
    name tdm
    version 1.0
    type tdm (temporal_modes=42, copies=3)

    int array a =
        1, 2

    int array b =
        3, 4

    Sgate(0.7, 0) | 1
    BSgate(a, 0.0) | [0, 1]
    MeasureHomodyne(phi=b) | 0
```

Will that produce a parsing error?

Reply:
Yes, it should. In that case I believe it would try to validate the type of `a` and `b` and fail (since they are arrays).